File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/abstracts/common/MacBorder.scala
package org.lang.scala.conclusion.design.factory.abstracts.common

/**
 * 	This is an abstract class to be used to test the implementation of abstract factory pattern
 *  -- This class must inherit a parent abstract class
 *  
 * 	@author VinceYuan
 */
class MacBorder extends Border {
  
	override def scroll() = {
	  println("A MAC Border is being scrolled...")
	}
}

File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/abstracts/common/MacBorder.java
package com.java.se.conclusion.design.factory.abstracts.common;

/**
 * 	This is an abstract class to be used to test the implementation of abstract factory pattern
 *  -- This class must inherit a parent abstract class
 *  
 * @author VinceYuan
 *
 */
public class MacBorder extends Border {

	@Override
	public void scroll() {
		System.out.println("A MAC Border is being scrolled...");
	}
}
